opentracing-agent/src/main/java/io/opentracing/contrib/agent/OpenTracingHelper.java

```diff
@@ -0,0 +1,285 @@
+/*
+ * Copyright 2017 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.opentracing.contrib.agent;
+
+import java.net.HttpURLConnection;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.WeakHashMap;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import org.jboss.byteman.rule.Rule;
+import org.jboss.byteman.rule.helper.Helper;
+
+import io.opentracing.Span;
+import io.opentracing.SpanContext;
+import io.opentracing.Tracer;
+import io.opentracing.Tracer.SpanBuilder;
+import io.opentracing.contrib.global.GlobalTracer;
+import io.opentracing.contrib.spanmanager.DefaultSpanManager;
+import io.opentracing.contrib.spanmanager.SpanManager;
+import io.opentracing.contrib.spanmanager.SpanManager.ManagedSpan;
+import io.opentracing.propagation.Format;
+
+/**
+ * This class provides helper capabilities to the byteman rules.
+ */
+public class OpenTracingHelper extends Helper {
+
+    private static final Logger log = Logger.getLogger(OpenTracingHelper.class.getName());
```

Comment:
While I agree on the common case, this here is a Java Agent. It is something that is started up before the application itself, so, it's not possible to know before hand what's the logging strategy for the target application. This effectively mean that if we need to log something before the application has been started, we might not have a log4j implementation ready to use yet.